src/urlschemehandler.cpp

```diff
@@ -13,30 +13,30 @@
 #include <zim/item.h>
 #include <zim/error.h>
 
-
-UrlSchemeHandler::UrlSchemeHandler()
+zim::Entry getArchiveEntryFromUrl(const zim::Archive& archive, const QUrl& url)
 {
+    std::string path = url.path().toUtf8().constData();
+    if (path[0] == '/')
+        path = path.substr(1);
+
+    try {
+        return archive.getEntryByPath(path);
+    } catch (zim::EntryNotFound& e) {
+        if (path.empty() || path == "/") {
+            return archive.getMainEntry();
+        }
+    }
+    throw zim::EntryNotFound("Cannot find entry for non empty path");
 }
 
-zim::Entry getEntryFromPath(const zim::Archive& archive, const std::string& path)
+UrlSchemeHandler::UrlSchemeHandler()
 {
-  try {
-    return archive.getEntryByPath(path);
-  } catch (zim::EntryNotFound& e) {
-    if (path.empty() || path == "/") {
-      return archive.getMainEntry();
-    }
-  }
-  throw zim::EntryNotFound("Cannot find entry for non empty path");
 }
```

Comment:
The placement of `getArchiveEntryFromUrl()` and the change in whitespace make the diff in this commit more confusing than it could be. History of code is as important as its most recent state. Please don't introduce unnecessary difficulties in understanding the changes in the code. Cleanup actions like moving code around for better cohesion etc are welcome but they should be performed as separate steps in commits of their own. A general rule - try to not combine changes of different nature in the same commit and try keep the size of each change within an average software engineer's ability to comprehend it in a single pass.

Reply:
Done, the change log does look confusing...